lib/services/apparmor.pm

```diff
@@ -0,0 +1,149 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Package for apparmor service tests
+#
+# Maintainer: Huajian Luo <hluo@suse.com>
+
+package services::apparmor;
+use base "apparmortest";
+use testapi;
+use utils;
+use version_utils 'is_sle';
+use warnings;
+use strict;
+
+sub install_service {
+    # apparmor need be installed by pattern
+    zypper_call 'in -t pattern apparmor';
```

Comment:
check with security lily for how to correctly install apparmor in 12sp2/3/4, 15 etc..

Reply:
I only can confirm AppAmror is a pattern on Tumbleweed, sle12sp4+ and sle15+. Not sure on other versions/products.